readme.md

```diff
@@ -64,7 +72,22 @@ The software changes required to support this consensus change differ significan
-Beyond dropping the unnecessary limit on VM number length, this proposal does not modify any other properties of the VM. Notably, the precise format and behavior of VM numbers across all VM operations – especially `OP_1ADD` (`0x8b`), `OP_1SUB` (`0x8c`), `OP_NEGATE` (`0x8f`), `OP_ABS` (`0x90`), `OP_NOT` (`0x91`), `OP_0NOTEQUAL` (`0x92`), `OP_ADD` (`0x93`), `OP_SUB` (`0x94`), `OP_MUL` (`0x95`), `OP_DIV` (`0x96`), `OP_MOD` (`0x97`), `OP_BOOLAND` (`0x9a`), `OP_BOOLOR` (`0x9b`), `OP_NUMEQUAL` (`0x9c`), `OP_NUMEQUALVERIFY` (`0x9d`), `OP_NUMNOTEQUAL` (`0x9e`), `OP_LESSTHAN` (`0x9f`), `OP_GREATERTHAN` (`0xa0`), `OP_LESSTHANOREQUAL` (`0xa1`), `OP_GREATERTHANOREQUAL` (`0xa2`), `OP_MIN` (`0xa3`), `OP_MAX` (`0xa4`), and `OP_WITHIN` (`0xa5`) – are part of network consensus and do not otherwise change as a result of this proposal. For the avoidance of doubt, see the [Tests & Benchmarks](#tests--benchmarks).
+As described in [Interaction with Existing Operations](#interaction-with-existing-operations), this proposal does not modify any other properties of the VM. Notably, the precise format and behavior of VM numbers across all VM operations is part of network consensus and does not change as a result of this proposal. For the avoidance of doubt, see the [Tests & Benchmarks](#tests--benchmarks).
+
+### Removal of Overflow Behavior
+
+As a practical consequence of removing the limit, numerical operation overflow is no longer a meaningful concept. A number that becomes too large would be too large due to the cost system established in [`CHIP: Targeted Virtual Machine Limits`](https://github.com/bitjson/bch-vm-limits), not due to a numerical boundary. These operations previously required overflow considerations: `OP_1ADD` (`0x8b`), `OP_1SUB` (`0x8c`), `OP_ADD` (`0x93`), `OP_SUB` (`0x94`), `OP_MUL` (`0x95`).
+
+Example showing previous 64-bit integer overflow no longer overflows:
+
+- In the current VM, multiplication of two large 64-bit numbers using `OP_MUL` (`0x95`), where the result is too large to fit in the 64-bit number specification, will result in an overflow and therefore script failure. In other words, the two inputs are valid, but the output is too large.
+- In the VM upgraded with both this CHIP and the cost system of [`CHIP: Targeted Virtual Machine Limits`](https://github.com/bitjson/bch-vm-limits), the multiplication of the same two numbers no longer results in script failure per se. It might result in script failure if the overall cost of the transaction happens to be exceeded by this specific operation.
+
+Example showing removal of overflow behavior in general:
+
+- In the VM upgraded with both this CHIP and the cost system of [`CHIP: Targeted Virtual Machine Limits`](https://github.com/bitjson/bch-vm-limits), the maximum number is a 10,000 byte number, due to the cost system.
+- Multiplying two valid numbers that result in, for example, a 10,005 byte number would result in script failure due to the result being too large to push onto the stack.
```

Comment:
and sure, implementations could opt to throw a stack item size error, but bchn one will throw `ScriptError::INVALID_NUMBER_RANGE_BIG_INT` on overflow

Reply:
Indeed I am confused. That's not a good sign.

I'm still not sure what the actual answer is. Before it sounded like there is a simple maxint that is enforced on.... outputs only?? Inputs only get size checked?

Reply:
I recommend resolving this thread since it's all connected to the conversation above.

Reply:
>Above it sounded like there is a simple maxint that is enforced on.... outputs only?? Inputs only get byte-size checked?

That's the effective result of having a stack item size limit. Result of 10,001 bytes MUST fail. Implementations can choose which error to throw, but they MUST throw something.

If you consistently enforce that **any** output of an opcode MUST be `<=` 10,000 bytes, then why would you need to check for input too - when it will not be possible to have a 10,001 item on stack, it becomes an assertion rather than enforcement: `assert(input_size <= 10,000)` if it fails then something went terribly wrong in your implementation (some opcode "leaked" through the output limit or stack state got corrupt or something)

Reply:
I'll stick with the other thread. This topic is encompassed there.